### ISSUE TITLE
Remove RISM Siglum/private institution dichotomy for institutions

### ISSUE THREAD

**Issue by dchiller** [DDMAL/CantusDB#1630]
Current options for an institution are "public" (ie. has a RISM siglum) and "private_collector". Does this constraint make sense?

**Comment by ahankinson** [comment]
Hm. I wonder if the institution model is the best place for this. 

Each part of a virtual reconstruction is usually also a "Source" in its own right. So it would have its own holding institution, etc. But every source can only have one holding institution. 

It seems to me that it would be better to have a way of joining together multiple sources into a virtual set. In DIAMM we have the `Set` model that allows us to group many sources together. We use it primarily for partbooks, but it can also be used for other purposes. 

For example the Worcester Fragments are gathered into a single set, even though they live in different institiutions:

https://www.diamm.ac.uk/sets/3/

Or the Egerton/Wolffheim chansonnier

https://www.diamm.ac.uk/sets/435/

These both have a set type of "Fragments of a whole".

**Comment by annamorphism** [comment]
Does a set allow ordering? Part of the reconstruction is not just having them grouped, but also deciding what order they should go in. 
Also, at the moment our lone published reconstructed source is not (as far as Cantus is concerned) made up of parts with their own holding institutions.  This could easily change, though--we already have some files where the chants are associated to the right folios from each holding institution, so if it could be reassembled by associating (and ordering) the constituent fragments, that could be the way for any future reconstructions.

**Comment by ahankinson** [comment]
> Does a set allow ordering? Part of the reconstruction is not just having them grouped, but also deciding what order they should go in.

It does if we say it does! 😁 

> Also, at the moment our lone published reconstructed source is not (as far as Cantus is concerned) made up of parts with their own holding institutions.

Could it be? I'm thinking that it's the Gottschalk Antiphoner, and it gives the library and shelfmark for each fragment: 

https://cantusdatabase.org/source/123609

(PS: The formatting on this page could use some love...)

> This could easily change, though--we already have some files where the chants are associated to the right folios from each holding institution, so if it could be reassembled by associating (and ordering) the constituent fragments, that could be the way for any future reconstructions.

This sounds like there's a bit more modelling needed for this, but it's certainly doable.

I just get a bit squirrely when I see data models being used for not-quite their intended purpose, and always think that modelling the data correctly is better in the long run, even if it means not being expedient about it in the short term.

**Comment by annamorphism** [comment]
> Could it be? I'm thinking that it's the Gottschalk Antiphoner, and it gives the library and shelfmark for each fragment:
> 
> https://cantusdatabase.org/source/123609

Exactly--there are (right now) no Cantus "sources" associated with each of those components, just the reconstruction (so the reconstruction effectively has no associated institutions, even though all the constituent pages have some sort of institution in real life). But source pages could be made for those fragments--they're a CSV upload away. 

> (PS: The formatting on this page could use some love...)

Yeah...it got wonkified somewhere in the migration to NewCantus and never quite got fixed...

**Comment by dchiller** [comment]
Right, the institution itself if not a virtual reconstruction. 

I think what we'll do here then is remove the requirement that an institution have a RISM siglum or be a private institution. There are institutions, such as those making reconstructions in the two cases we have (Gottschalk already and Beauvais in progress), that neither have a siglum nor are private.

Sets sound like a good idea, though they don't solve the immediate point about institutions and source references for reconstructions. 

It might be a good idea to be able to mark a source (for now) as a reconstruction -- I could see something like `(reconstruction)` being used similar to how we use `(fragment)` in the Cantus siglum field.

**Comment by ahankinson** [comment]
> I think what we'll do here then is remove the requirement that an institution have a RISM siglum or be a private institution. There are institutions, such as those making reconstructions in the two cases we have (Gottschalk already and Beauvais in progress), that neither have a siglum nor are private.

I pretty strongly disagree with this. Those examples are not institutions, so why model them as such? What does it bring? If there is no need to capture a place or any institutional information, then changing the institution model constraints for everything, just to meet the needs of this small use case, seems like the wrong direction.

I would suggest removing the requirement that a source has a holding institution before changing the behaviour of the Institution model. If it doesn’t have a holding institution, then there shouldn’t be a need to “hack” something in place.

Even better would be to remove the requirement for a holding institution only if the source record is set to be a reconstruction. 

The point is that there are ways to do this without needing to modify the institution model, which is ill suited for this particular use case anyway.

**Comment by annamorphism** [comment]
I think this issue is no longer relevant? closing.
